### ISSUE TITLE
Possible bug in the calculating score in score_utils scores_vs_time function

### ISSUE THREAD

**Issue by bannanc** [open-forcefield-group/smarty#152]
I've been using the score_utils functions to look at how well smirky is doing. I think there is a bug in how scores_vs_time is computed. ( @davidlmobley I think you wrote the original version of these?) However I changed the `load_trajectory` function to use pandas instead so I don't know if that introduced a bug for some reason. @camizanette Have you seen this in your smarty data analysis at all?

I've been using an interactive python window to reach in trajectory files and then look for the iteration with the highest score. Then I find that iteration in the log file to look at the SMIRKS patterns being created. However, in my most recent Angle simulation, it looks like there is a bug with the computed score. 

the scores_vs_time function returns a dictionary with the form:
keys: 'all', 'NONE', and all reference types ('a0001' to 'a0006' for AlkEtOH angles)
entries: numpy arrays with score for that key at the iteration matching that index

For my recent Angle simulation the max under all happened at iteration 298 with a score of 99.2%
However the score in the log file is only 75.784%.
For all the reference types the score from the `scores_vs_time` dictionary are correct so I'm concerned there is an error in how the 'all' score is being calculated. I've looked at the `scores_vs_time` function, but I can't identify what would be causing this bug. 

Here is the data for that entry in the log file:

```
INDEX        ANGLES  MOLECULES   TYPE NAME: SMIRKS                                  REF TYPE: SMIRKS                                   FRACTION OF REF TYPED MOLECULES MATCHED
    1 :        164         42 | C: [*:1]~[#6:2]~[*:3]                              a0002: [#1:1]-[#6X4:2]-[#1:3]                   164 /     164 (100.000%)
    2 :         42         30 | O: [*:1]~[#8:2]~[*:3]                              a0006: [#6X4:1]-[#8X2:2]-[#6X4:3]                       39 /      39 (100.000%)
    3 :         65         42 | 1925: [#6:1](~[#6])~[#8:2]~[#1:3]                  a0005: [#6X4:1]-[#8X2:2]-[#1:3]                         65 /      68 ( 95.588%)
    4 :        160         40 | 3915: [#6:1]~[#6:2]~[*:3]                          a0003: [#6X4:1]-[#6X4:2]-[#6X4:3]                      160 /     160 (100.000%)
    5 :        669         42 | 3274: [#1:1]~[#6:2]-[#6:3]                         a0001: [a,A:1]-[#6X4:2]-[a,A:3]                        669 /    1029 ( 65.015%)
    6 :        169         30 | 8590: [#8:1]~[#6:2]~[*:3]
    7 :         17         14 | 7055: [#6:1](~[#6:2]~[#8:3])~[#8]
    8 :        168         42 | 4935: [#8:1](~[#1])~[#6:2]~[*:3]
    9 :         39         22 | 3367: [#8:1]~[#6:2]~[#8:3]                         a0004: [#8X2:1]-[#6X4:2]-[#8X2:3]                       39 /      39 (100.000%)
   10 :          6          6 | 8497: [#6:1](~[#6:2]~[#8:3])(~[#8])~[#8]
TOTAL :       1499         42 |                                                       1136 /     1499 match (75.784 %)
```

And from the trajectory file:

```
298,1,'[*:1]~[#6:2]~[*:3]',0,0,'a0002',164,42,164,164
298,2,'[*:1]~[#8:2]~[*:3]',0,0,'a0006',42,30,39,39
298,3,'[#6:1](~[#6])~[#8:2]~[#1:3]',0,0,'a0005',65,42,65,68
298,4,'[#6:1]~[#6:2]~[*:3]',0,0,'a0003',160,40,160,160
298,5,'[#1:1]~[#6:2]-[#6:3]',0,0,'a0001',669,42,669,1029
298,6,'[#8:1]~[#6:2]~[*:3]',0,0,'NONE',169,30,0,0
298,7,'[#6:1](~[#6:2]~[#8:3])~[#8]',0,0,'NONE',17,14,0,0
298,8,'[#8:1](~[#1])~[#6:2]~[*:3]',0,0,'NONE',168,42,0,0
298,9,'[#8:1]~[#6:2]~[#8:3]',0,0,'a0004',39,22,39,39
298,10,'[#6:1](~[#6:2]~[#8:3])(~[#8])~[#8]',0,0,'NONE',6,6,0,0
```

**Comment by bannanc** [comment]
I've addressed this in pull request #151 
for trajectory files with the "total" line entry. @camizanette we will need to add the total line to the trajectory files in smarty to guarantee this same bug isn't happening there. 

The bug stemmed from the fact that the trajectory files don't necessarily have all reference types at each iteration.

**Comment by bannanc** [comment]
For smirky this is fixed with pull request #151 
I gave the changes for the sampler to @camizanette to fix smarty and smarty_elemental. Camila, when those are updated could you close this?

**Comment by bannanc** [comment]
I fixed these in pull request #151 There are more issues with the elemental sampler, but those are listed in issue #202
